From 155e5225a89764f8cd0c47bc598e653a641f5fab Mon Sep 17 00:00:00 2001
From: Pavel Dubrova <pashadubrova@gmail.com>
Date: Mon, 20 Dec 2021 22:06:53 +0200
Subject: [PATCH 083/175] clk: qcom: gcc-sdm845: Reorder DFS clock registration

DFS wrap0/1 SE clock registration must be done before
qcom_cc_really_probe() is executed to properly initialize
their child branch clocks.

Signed-off-by: Pavel Dubrova <pashadubrova@gmail.com>
---
 drivers/clk/qcom/gcc-sdm845.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/clk/qcom/gcc-sdm845.c b/drivers/clk/qcom/gcc-sdm845.c
index c39070fecb2d..aeca2812675b 100644
--- a/drivers/clk/qcom/gcc-sdm845.c
+++ b/drivers/clk/qcom/gcc-sdm845.c
@@ -4336,6 +4336,14 @@ static int gcc_sdm845_probe(struct platform_device *pdev)
 	if (ret)
 		return ret;
 
+	/* DFS clock registration */
+	ret = qcom_cc_register_rcg_dfs(regmap, gcc_dfs_clocks,
+			ARRAY_SIZE(gcc_dfs_clocks));
+	if (ret) {
+		dev_err(&pdev->dev, "Failed to register with DFS!\n");
+		return ret;
+	}
+
 	ret = qcom_cc_really_probe(pdev, &gcc_sdm845_desc, regmap);
 	if (ret) {
 		dev_err(&pdev->dev, "Failed to register GCC clocks\n");
@@ -4350,13 +4358,8 @@ static int gcc_sdm845_probe(struct platform_device *pdev)
 	if (!of_device_is_compatible(pdev->dev.of_node, "qcom,gcc-sdm845-v2.1"))
 		clk_prepare_enable(gcc_aggre_noc_pcie_tbu_clk.clkr.hw.clk);
 
-	/* DFS clock registration */
-	ret = qcom_cc_register_rcg_dfs(regmap, gcc_dfs_clocks,
-			ARRAY_SIZE(gcc_dfs_clocks));
-	if (ret)
-		dev_err(&pdev->dev, "Failed to register with DFS!\n");
-
 	dev_info(&pdev->dev, "Registered GCC clocks\n");
+
 	return ret;
 }
 
-- 
2.48.1

